Assets/Scripts/Battlefield/TurnMechanism/TurnManager.cs

```diff
@@ -1,42 +1,48 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
-using SwordAndBored.TurnMechanism;
+using SwordAndBored.Battlefield.TurnMechanism;
 using UnityEngine.UI;
 using TMPro;
+using SwordAndBored.Battlefield.CreaturScripts;
 
-public class TurnManager : MonoBehaviour
+namespace SwordAndBored.Battlefield.TurnMechanism
 {
-    public GameObject[] units;
-    TurnOrderController manager;
-    public UniqueCreature activePlayer;
-    public TextMeshProUGUI text;
-
-    public Image[] actionsLeft;
-
-    void Start()
+    public class TurnManager : MonoBehaviour
     {
-        manager = new TurnOrderController(units, new RandomShuffler<GameObject>());
-        activePlayer = manager.NextEntity().GetComponent<UniqueCreature>();
-        text.text = "Current Player: " + activePlayer.gameObject.name;
-    }
+        [Header("Units")]
+        public GameObject[] units;
+        public UniqueCreature activePlayer;
+        [Header("UI")]
+        TurnOrderController manager;
+        public TextMeshProUGUI text;
 
-    public void nextTurn()
-    {
-        activePlayer = manager.NextEntity().GetComponent<UniqueCreature>();
-        text.text = "Current Player: " + activePlayer.gameObject.name;
-        activePlayer.StartTurn();
-    }
+        public Image[] actionsLeft;
 
-    void Update()
-    {
-        if (activePlayer.action)
+        void Start()
         {
-            actionsLeft[0].color = new Color(0, 0, 1, .2f);
+            manager = new TurnOrderController(units, new RandomShuffler<GameObject>());
+            activePlayer = manager.NextEntity().GetComponent<UniqueCreature>();
+            text.text = "Current Player: " + activePlayer.creatureName;
         }
-        else
+
+        public void nextTurn()
+        {
+            activePlayer = manager.NextEntity().GetComponent<UniqueCreature>();
+            text.text = "Current Player: " + activePlayer.creatureName;
+            activePlayer.StartTurn();
+        }
+
+        void Update()
         {
-            actionsLeft[0].color = new Color(1, 0, 0, .2f);
+            if (activePlayer.action)
```

Comment:
This logic should be in the player himself and there should be something akin to activePlayer.doTurn() in here and the player holds his own state information.

Reply:
The UI changes like this are going to be a part of what Rory is working on.  For now we can leave this here and Rory's UI branch will fix this and render color changes like this unnecessary